### DISCUSSION TITLE
Feature Request: parquet over ssh

### DISCUSSION THREAD

**Discussion by gowerc** [duckdb/duckdb#14798] [Ideas]
At my company we often store data on a remote server that we access via ssh in parquet format. Duckdb is awesome for querying and manipuling parquet files but we currently have to make a local copy of the data before we can use it. Would be good if the read_parquet function supported ssh reading.

**Comment by soerenwolfers** [comment]
Are you using SFTP? That and many other protocols are already supported in the Python client via fsspec, see https://duckdb.org/docs/guides/python/filesystems.html

**Reply by gowerc** [reply]
I think so but will double check. That being said we are mostly an R shop. Do you have an equivalent guide for R?

**Reply by soerenwolfers** [reply]
This functionality doesn't exist in the R client

**Reply by soerenwolfers** [reply]
You could also consider hosting the files via HTTPS or S3 which are supported everywhere https://duckdb.org/docs/extensions/httpfs/https.html

**Comment by fanyang01** [comment]
Hi @gowerc,

I'm working on an open-source project that runs DuckDB in server mode: https://github.com/apecloud/myduckserver . It opens a Postgres-compatible port where you can send DuckDB queries (like scanning Parquet files) and retrieve the results. I’m not sure if it aligns with your use cases, but let me know if it could be helpful or if there’s anything we could improve!

**Comment by nbc** [comment]
Of you use Linux  you can try sshfs to mount remote directories over sftp

https://github.com/libfuse/sshfs

**Comment by onnimonni** [comment]
I created https://github.com/midwork-finds-jobs/duckdb-sshfs few weeks ago which provides this exact functionality. It's also available in the community extensions. Let me know if it works for you 😊
